src/Workspaces/Core/Portable/Workspace/Workspace_Events.cs

```diff
@@ -78,14 +78,19 @@ protected Task RaiseWorkspaceChangedEventAsync(WorkspaceChangeKind kind, Solutio
             projectId = documentId.ProjectId;
         }
 
-        var args = new WorkspaceChangeEventArgs(kind, oldSolution, newSolution, projectId, documentId);
```

Comment:
is saving making the args worth it?

Reply:
Probably not a huge difference, but the creation of those args caused a separate issue for another team's test that called this with a null oldSolution. We had them changed that, but only creating the args when necessary would have avoided the issue for them. And this is actually how I had originally coded this up before code reviews for the original change, so I thought it best to go back to something I had tested more thoroughly.